src/invidious/yt_backend/youtube_api.cr

```diff
@@ -488,6 +492,9 @@ module YoutubeAPI
           "html5Preference": "HTML5_PREF_WANTS",
         },
       },
+      "serviceIntegrityDimensions" => {
+        "poToken" => CONFIG.po_token ? CONFIG.po_token.as(String) : nil,
```

Comment:
I have used a ternary operator because I had issues with types when trying to pass po_token from an if case after the object was created.

Reply:
Using a ternary is fine but if you'd like to avoid that then you'll need to assign an intermediate variable for the compiler to recognize the type.

```crystal
if po_token = CONFIG.is_a?(String)
  # here po_token is guaranteed to be a String
end
````

See https://crystal-lang.org/reference/1.13/syntax_and_semantics/if_varis_a.html

Reply:
Even better: given the type specified in `config.cr` (which is `String | Nil`) then you can pass that value directly, no need for anycomplex logic! If `po_token` is a `String` a string will be put in here, if it's `Nil` then a `nil` will be put here.

Reply:
```suggestion
        "poToken" => CONFIG.po_token,
```